lib/auth.ts

```diff
@@ -59,7 +61,7 @@ export const verifyServerSideAuthorization = async (context: Context) => {
     res.writeHead(302, {
       Location: `https://stagingapi.artsy.net/oauth2/authorize?client_id=${
         process.env.APP_ID
-      }&redirect_uri=http://${host}/oauth2/callback?redirect_to=${encodeURI(
+      }&redirect_uri=${urlFromReq(req)}/oauth2/callback?redirect_to=${encodeURI(
```

Comment:
We had a whole slew of URLs that only worked on `http` or `localhost`. This is a unifying way of fixing those.